### PR TITLE
interop, doc: add interop test descrption for api_listener, metadata_filter, forwarding_rule_port_match, forwarding_rule_default_port

### PR DESCRIPTION
**Pull request by YifeiZhuang**
Related to https://github.com/grpc/grpc/pull/26065

### CODE REVIEW

File: doc/xds-test-descriptions.md [hunk 1]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+
+Assert:
+
+Test driver configures and verifies:
+1. No traffic goes to backends when configuring the target URI 
+`xds:///myservice`, the forwarding rule with port *x != 80*, the URL map 
+host rule `myservice::x`.
+1. Traffic goes to backends when configuring the target URI `xds:///myservice`, 
+the forwarding rule port `80` and the URL map host rule `myservice`.
+1. Not traffic goes to backends when configuring the target URI
```
Comment (ericgribkoff) [inline doc/xds-test-descriptions.md:728]: s/Not/No

File: doc/xds-test-descriptions.md [hunk 2]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+configuration while keep using the existing name.
+
+Client parameters:
+
+1.  --num_channels=1
+1.  --qps=100
+
+Load balancer configuration:
+
+1.  One MIG with each having two backends.
```
Comment (ericgribkoff) [inline doc/xds-test-descriptions.md:638]: s/each having// (here and elsewhere)

File: doc/xds-test-descriptions.md [hunk 3]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+URL map with the same test host name.
+1. The test driver deletes the first set of configurations in step 1.
+
+The test driver verifies, at each configuration step, the traffic is always able 
+to reach the designated hosts.
+
+### metadata_filter
+This test case verifies that metadata filter configurations in URL map match 
+rule are effective at Traffic Director for routing selection against downstream
+ node metadata.
```
Comment (ericgribkoff) [inline doc/xds-test-descriptions.md:655]: Nit: remove leading space?

File: doc/xds-test-descriptions.md [hunk 4]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+There are four test sub-cases:
+1. Test `MATCH_ALL` metadata filter criteria.
+1. Test `MATCH_ANY` metadata filter criteria.
+1. Test mixed `MATCH_ALL` and `MATCH_ANY` metadata filter criteria.
+1. Test when multiple match rules with metadata filter all match.
+
+Assert:
+
+At each test sub-case described above, the test driver configures
+ and verifies:
```
Comment (ericgribkoff) [inline doc/xds-test-descriptions.md:675]: Nit: leading space

File: doc/xds-test-descriptions.md [hunk 5]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+1.  --num_channels=1
+1.  --qps=100
+
+Load balancer configuration:
+
+1.  One MIG with each having two backends.
+
+Assert:
+1. The test driver configures matching port in the forwarding rule and in the
+ request server uri, then verifies traffic reaches backend service instances.
```
Comment (ericgribkoff) [inline doc/xds-test-descriptions.md:700]: Nit: leading space

**Review by ericgribkoff — APPROVED** [review]
Thanks for doing this!

File: doc/xds-test-descriptions.md [hunk 6]
```diff
@@ -621,3 +621,110 @@ There are four sub-tests:
...
+1. Test mixed `MATCH_ALL` and `MATCH_ANY` metadata filter criteria.
+1. Test when multiple match rules with metadata filter all match.
+
+Assert:
+
+At each test sub-case described above, the test driver configures
+and verifies:
+
+1. Set default URL map, and verify traffic goes to the original backend hosts. 
+1. Then patch URL map to update the match rule with metadata filter
```
Comment (ejona86) [inline doc/xds-test-descriptions.md:678]: It seems this should say "_add_ a match rule _for the alternate MIG_". And (1) should say "Set a default URL map pointing to one of the MIGs"

It isn't at all clear what configuration is expected here. Is this only positive testing? As in, let's say that the server ignored the metadata match criteria (essentially becoming wildcard), wouldn't this test still pass? What filter labels are to be used? Only a single label, or multiple labels? Is it okay if all labels match for the MATCH_ANY?

**Pull request merged**